dlc-oracle/src/main/scala/org/bitcoins/dlc/oracle/DLCOracle.scala

```diff
@@ -0,0 +1,206 @@
+
+  def stakingAddress(network: BitcoinNetwork): Bech32Address =
+    Bech32Address(P2WPKHWitnessSPKV0(signingKey.publicKey), network)
+
+  protected[bitcoins] val rValueDAO: RValueDAO = RValueDAO()
+  protected[bitcoins] val eventDAO: EventDAO = EventDAO()
+  protected[bitcoins] val eventOutcomeDAO: EventOutcomeDAO = EventOutcomeDAO()
+
+  private def getPath(keyIndex: Int): BIP32Path = {
+    val accountIndex = rValueAccount.index
+    val coin = rValueAccount.coin
+    val purpose = coin.purpose
+
+    BIP32Path.fromString(
+      s"m/${purpose.constant}'/${coin.coinType.toInt}'/$accountIndex'/$chainIndex'/$keyIndex'")
+  }
+
+  private def getKValue(rValDb: RValueDb): ECPrivateKey =
+    getKValue(rValDb.label, rValDb.path)
+
+  private def getKValue(label: String, path: BIP32Path): ECPrivateKey = {
+    require(path.forall(_.hardened),
+            s"Cannot use a BIP32Path with unhardened nodes, got $path")
+    val priv = extPrivateKey.deriveChildPrivKey(path).key
+    val hash =
+      CryptoUtil.sha256(
+        priv.schnorrNonce.bytes ++ CryptoUtil.serializeForHash(label))
+    val tweak = ECPrivateKey(hash.bytes)
+
+    priv.add(tweak)
+  }
+
+  def listEventDbs(): Future[Vector[EventDb]] = eventDAO.findAll()
+
+  def listPendingEventDbs(): Future[Vector[EventDb]] = eventDAO.getPendingEvents
+
+  def listEvents(): Future[Vector[Event]] = {
+    for {
+      eventDbs <- eventDAO.findAll()
+      outcomes <- eventOutcomeDAO.findAll()
+    } yield {
+      val outcomesByNonce = outcomes.groupBy(_.nonce)
+      eventDbs.map(db => Event(db, outcomesByNonce(db.nonce)))
+    }
+  }
+
+  def createNewEvent(
+      label: String,
+      outcomes: Vector[String]): Future[EventDb] = {
+    for {
+      indexOpt <- rValueDAO.maxKeyIndex
+      index = indexOpt match {
+        case Some(value) => value + 1
+        case None        => 0
+      }
+
+      path = getPath(index)
+      nonce = getKValue(label, path).schnorrNonce
+
+      hash =
+        CryptoUtil.sha256(nonce.bytes ++ CryptoUtil.serializeForHash(label))
+      commitmentSig = signingKey.schnorrSign(hash.bytes)
+
+      rValueDb = RValueDbHelper(nonce = nonce,
+                                label = label,
+                                account = rValueAccount,
+                                chainType = chainIndex,
+                                keyIndex = index,
+                                commitmentSignature = commitmentSig)
+
+      eventDb = EventDb(nonce, label, outcomes.size, Mock, None)
+      eventOutcomeDbs = outcomes.map { outcome =>
+        val hash = CryptoUtil.sha256(outcome)
+        EventOutcomeDb(nonce, outcome, hash)
+      }
+
+      _ <- rValueDAO.create(rValueDb)
+      eventDb <- eventDAO.create(eventDb)
+      _ <- eventOutcomeDAO.createAll(eventOutcomeDbs)
+    } yield eventDb
+  }
+
+  def signEvent(
+      nonce: SchnorrNonce,
+      outcome: String): Future[SchnorrDigitalSignature] = {
```

Comment:
nit: I would maybe return the `EventDb` rather than `SchnorrDigitalSignature`. Realistically users of the API are going to probably want to know more information about the Event (and the `EventDb` should now contain `SchnorrDigitalSignature`).